pycargoebuild/cargo.py

```diff
@@ -63,11 +63,29 @@ class GitCrate(Crate):
 
     @property
     def download_url(self) -> str:
-        return f"{self.repository}/archive/{self.commit}.tar.gz"
+        if self.repository.startswith("https://github.com/"):
+            return f"{self.repository}/archive/{self.commit}.tar.gz"
+        if self.repository.startswith("https://gitlab."):
```

Comment:
Honestly, this is very, very confusing and took me around 10 minutes to notice what you're doing here. Also, I feel like we're copying the `.startswith()` check around too much.

So here's my idea: let's do the check once in `__init__` and store the repo type somewhere in the class as an enum (e.g. `GITHUB`, `GITLAB`, `GITLAB_SELFHOSTED`).